### ISSUE TITLE
*Unknown error SIGN_IN_REQUIRED(4)

### ISSUE THREAD

**Issue by Dipayon3D** [cranberrygame/cordova-plugin-game#30]
The plugin was working perfectly before...but all of a sudden it stopped working and the message in the title is showing the first time the app is opened.... After that it fails every time.... It's not even trying to connect.....It fails as soon as Button is clicked.....

**Comment by cruzlutor** [comment]
Same here, any clue?

**Comment by cruzlutor** [comment]
I found the solution to my problem, this plugin works if you use it on a signed apk build, also I had to link my app to my google console credentials

And dont forget to add testers accounts in your app settings

now is working

**EDIT**
Sometimes it works and some do not, so weird

**Comment by Dipayon3D** [comment]
Yes...I am having the same problem.... It's working sometimes, sometimes not.... But on some devices it's always working....!!! I don't understand.....

**Comment by TombasOnly** [comment]
Hi guys, I have same problem. It didn't work on my phone. So I delete my Google Play Games profile; delete app; install again; create new Google Play Games profile and now it works on my phone again :)

**Comment by fridadwi** [comment]
https://developers.google.com/games/services/console/enabling#c_specify_client_id_settings

I use those step and it works

**Comment by sajmons** [comment]
It only works in RELEASE mode. DEBUG mode will not work.

**Comment by Dipayon3D** [comment]
@sajmons my game is in RELEASE mode . . . i've set up everything correctly on google play dev console but it's still showing me this....

**Comment by sajmons** [comment]
@Dipayon3D check your config.xml if you included plugin with APP_ID parameter like this:

```
<plugin name="cordova-plugin-game" spec="^1.0.120">
        <variable name="APP_ID" value="your_game_services_id" />
</plugin>

```
And your package.json like this:

```
"cordova": {
    "platforms": [
      "android"
    ],
    "plugins": {
      "cordova-custom-config": {},
      "cordova-plugin-file": {},
      "cordova-plugin-inappbrowser": {},
      "cordova-plugin-statusbar": {},
      "cordova-plugin-whitelist": {},
      "cordova-plugin-fullscreen": {},
      "cordova-plugin-splashscreen": {},
      "cordova-plugin-x-socialsharing": {},
      "cordova-plugin-game": {
        "APP_ID": "your_game_services_id"
      }
    }
  } 
```

**Comment by Dipayon3D** [comment]
@sajmons the plugin is working sometimes . . . sometimes it's not working, showing me that *Unknown error SIGN_IN_REQUIRED(4) alert dialog . . . i think when the plugin fails to communicate with Google play, it throws that error... but the plugin IS WORKING - sometimes - that's what bothers me . . . this is a great plugin and it's unfortunate that this issue is making me look for workarounds.... did u not face this issue ever ??

**Comment by sajmons** [comment]
That's really strange. I don't have those problems.
How do you build your app? Does same builds works or not works every time? Maybe you have building issues that causing this?

**Comment by Samleo8** [comment]
Exactly the same issue for me, except mine doesn't work at all..
It's Cordova@8.0.0.

The app is released, signed and configured properly with Google Play Console.

The only build errors are:
`Parallel execution with configuration on demand is an incubating feature.
The Task.leftShift(Closure) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use Task.doLast(Action) instead.
	at build_8ban8a3bc0d6zeo7uf4zmez1o.run(/Users/sam/Documents/Mobile Apps/SoaringSheep/platforms/android/build.gradle:141)
The JavaCompile.setDependencyCacheDir() method has been deprecated and is scheduled to be removed in Gradle 4.0.
Incremental java compilation is an incubating feature.
The TaskInputs.source(Object) method has been deprecated and is scheduled to be removed in Gradle 4.0. Please use TaskInputs.file(Object).skipWhenEmpty() instead.`

Is there a workaround?
